src/components/Main.tsx

```diff
@@ -0,0 +1,687 @@
+  React.useEffect(() => {
+    logEvent('AppStart')
+  }, [])
+
+  return (
+    <NavigationContainer>
+      <Stack.Navigator
+        initialRouteName="login"
+        screenOptions={{
+          headerShown: false
+        }}
+      >
+        <Stack.Screen name="login" component={withServices(LoginScene)} />
+        <Stack.Screen name="edgeApp" component={EdgeApp} />
+      </Stack.Navigator>
+    </NavigationContainer>
+  )
+}
+
+const EdgeApp = () => {
+  const backPressedOnce = React.useRef(false)
+  const dispatch = useDispatch()
+  const navigation = useNavigation<NavigationBase>()
+
+  useBackEvent(() => {
+    if (backPressedOnce.current) {
+      return true
+    } else {
+      backPressedOnce.current = true
+      Airship.show(bridge => <AirshipToast bridge={bridge} message={s.strings.back_button_tap_again_to_exit} />).then(() => {
+        backPressedOnce.current = false
+      })
+      // Timeout the back press after 3 seconds so the state isn't "sticky"
+      setTimeout(() => {
+        backPressedOnce.current = false
+      }, 3000)
+      return false
+    }
+  })
+
+  useUnmount(() => {
+    dispatch(logoutRequest(navigation))
+  })
+
+  return (
+    <Drawer.Navigator
+      drawerContent={props => ifLoggedIn(ControlPanel)(props)}
+      initialRouteName="edgeAppStack"
+      screenOptions={{
+        drawerPosition: 'right',
+        drawerType: 'front',
+        drawerStyle: { backgroundColor: 'transparent', bottom: 0 },
+        headerShown: false
+      }}
+    >
+      <Drawer.Screen name="edgeAppStack" component={EdgeAppStack} />
+    </Drawer.Navigator>
+  )
+}
+
+const EdgeAppStack = () => {
+  const dispatch = useDispatch()
+
+  return (
+    <Stack.Navigator
+      initialRouteName="edgeTabs"
+      screenOptions={{
+        title: '',
+        headerTitle: ({ children }: HeaderTitleProps) => <HeaderTitle title={children} />,
+        headerLeft: () => <BackButton />,
+        headerRight: () => <SideMenuButton />,
+        headerShown: true,
+        headerTransparent: true
+      }}
+    >
+      <Stack.Screen
+        name="edgeTabs"
+        component={EdgeTabs}
+        options={{
+          headerShown: false
+        }}
+      />
+      <Stack.Screen name="createWalletSelectCrypto" component={ifLoggedIn(CreateWalletSelectCryptoScene)} />
+      <Stack.Screen name="createWalletSelectFiat" component={ifLoggedIn(CreateWalletSelectFiatScene)} />
+      <Stack.Screen
+        name="settingsOverview"
+        component={ifLoggedIn(SettingsScene)}
+        options={{
+          title: s.strings.title_settings
+        }}
+        listeners={{
+          focus: () => dispatch(showReEnableOtpModal())
+        }}
+      />
+      <Stack.Screen
+        name="transactionDetails"
+        component={ifLoggedIn(TransactionDetailsScene)}
+        options={{
+          headerTitle: () => <TransactionDetailsTitle />
+        }}
+        listeners={{
+          focus: async () => requestPermission('contacts')
```

Comment:
We don't await here, so why `async`? Maybe we want a `.catch(showError)` on the end, and then the function can return `void` instead of `Promise<void>`.

Reply:
This was added by the linter.